components/TokenApprovalList/RevokeButton.tsx

```diff
@@ -62,6 +70,9 @@ const RevokeButton: React.FC<Props> = ({ setAlert, listItem, account, hideItem }
   const currentContract = mapEthTypeToABI(listItem)
   const [callWrite, setCallWrite] = useState(false)
   const [isPackaging, setIsPackaging] = useState(false)
+  const { transaction_hash, spender_address_hash, token_contract_address_hash, data } = listItem
+  const itemKey = transaction_hash + spender_address_hash + token_contract_address_hash + data + '-revokeTxn'
+  const [revokeTxnHash, setRevokeTxnHash] = useState(localStorage.getItem(itemKey) || '')
```

Comment:
What if multiple approvals are revoked?

Reply:
this is inside component RevokeButton, so each button has its own state, did I miss anything?

Reply:
> this is inside component RevokeButton, so each button has its own state, did I miss anything?

It's me misunderstood it